server/src/main/java/io/deephaven/server/table/inputtables/InputTableServiceGrpcImpl.java

```diff
@@ -96,13 +97,18 @@ public void addTableToInputTable(
                         }
 
                         // actually add the tables contents
-                        try {
-                            inputTableUpdater.add(tableToAdd);
-                            GrpcUtil.safelyComplete(responseObserver, AddTableResponse.getDefaultInstance());
-                        } catch (IOException ioException) {
-                            throw Exceptions.statusRuntimeException(Code.DATA_LOSS,
-                                    "Error adding table to input table");
-                        }
+                        inputTableUpdater.addAsync(tableToAdd, new InputTableStatusListener() {
```

Comment:
Do we have to worry about the async calls throwing?

Reply:
No - the ExportBuilder has onError handler that plumbs any unexpected exceptions as errors to the response observer.